.github/workflows/test-rust.yaml

```diff
@@ -0,0 +1,75 @@
+name: Test the rust code
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  build_and_cache_rust:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+      - name: Update local toolchain
+        run: |
+          rustup update
+          rustup component add clippy rustfmt
+          rustup install stable          
+      - name: Toolchain info
+        run: |
+          cargo --version --verbose
+          rustc --version
+      - name: Fetch cargo deps
+        run: cargo fetch
+      - name: Run cargo build
+        run: cargo build
+      - name: Cache deps
```

Comment:
Might be better to use https://github.com/Swatinem/rust-cache

Reply:
Didn't know about this actions so replaced it!